### PR TITLE
reducedDims elements can be matrix-like objects

### PR DESCRIPTION
**Pull request by drisso**

### CODE REVIEW

File: tests/testthat/test-sce-subset.R [hunk 1]
```diff
@@ -1,22 +1,28 @@
 # Checks the subsetting methods.
 # library(SingleCellExperiment); library(testthat); source("test-sce-subset.R")
+context("SCE subsetting")
 
 set.seed(1000)
 ncells <- 100
 v <- matrix(rnorm(20000), ncol=ncells)
 pca <- matrix(runif(ncells*5), ncells)
-sce <- SingleCellExperiment(assay=v, reducedDims=SimpleList(PCA=pca)) # A fully loaded object.
```
Comment (LTLA) [inline tests/testthat/test-sce-subset.R]: Duplicate the test rather than replacing it with an LEM example.

Comment (LTLA) [reply tests/testthat/test-sce-subset.R]: Wait, I see what you've done. In that case, duplicate the test in the new test file (`test-sce-with-lem.R`) that you've created. Isolate LEM+SCE tests from the SCE-only tests to a single file.

Comment (LTLA) [reply tests/testthat/test-sce-subset.R]: Also test `cbind` for SCEs (i.e., `rbind` LEMs). This is actually less obvious than it seems, as it only makes sense to `rbind` LEMs that have the same loading vectors; should probably check that in `rbind,LEM-method`.

**Comment by drisso** [comment]
What about now?

**Comment by LTLA** [comment]
You seem to be missing some `cbind` tests.

**Comment by drisso** [comment]
I've added `cbind` tests. Have a look at https://github.com/drisso/SingleCellExperiment/blob/develop/tests/testthat/test-sce-with-lem.R#L82.

Is this the expected behavior or should we implement a `LinearEmbeddingMatrix` + `DataFrame` cbind?

**Comment by LTLA** [comment]
No, I think it's fine. I don't think it makes sense to `rbind` a LEM with a non-LEM. I'm not even sure we can do it, because the `rbind` generic seems to require all objects to be of the same class.

**Pull request merged**